src/main.rs

```diff
@@ -189,15 +265,46 @@ fn create_swarm(identity: identity::Keypair) -> Result<Swarm<Behaviour>> {
     Ok(swarm)
 }
 
-fn create_transport(identity: &identity::Keypair) -> Result<Boxed<(PeerId, StreamMuxerBox)>> {
+fn create_transport(
+    identity: &identity::Keypair,
+    websocket: bool,
+    tls: Option<tls::Config>,
+) -> Result<Boxed<(PeerId, StreamMuxerBox)>> {
+    let tcp_with_dns = TokioDnsConfig::system(TokioTcpConfig::new().nodelay(true)).unwrap();
+
+    let transport = if websocket {
+        let mut websocket_with_dns = WsConfig::new(tcp_with_dns.clone());
+
+        if let Some(tls) = tls {
+            websocket_with_dns.set_tls_config(tls);
+        }
+
+        authenticate_and_multiplex(
+            tcp_with_dns.or_transport(websocket_with_dns).boxed(),
+            &identity,
+        )
+        .unwrap()
+    } else {
+        authenticate_and_multiplex(tcp_with_dns.boxed(), &identity).unwrap()
+    };
```

Comment:
Not sure why you are suddenly unwrapping in here when the fn actually returns a `Result`.